### ISSUE TITLE
Mouse Over Elements

### ISSUE THREAD

**Issue by fisher729** [nvaccess/nvda#6048]
NVDA should have the ability to anounce elements with an on mouse over attribute. This should be a setting that can be toggled. The user would then have to know that the page should be browsed, since there's no way in NVDA that the text can be jumped to automatically.

**Comment by fisher729** [comment]
...be jumped to automatically.

**Comment by jcsteh** [comment]
I thought we already had an issue for this, but if we do, I can't find it either. :)

There are several problems here:

1. A sighted user doesn't know that moving the mouse over something will cause something to happen either. That is, the presence of an onMouseOver attribute doesn't create any visual indication. Given that, it seems incorrect for a screen reader to provide such an indication. The page might use an icon or the like, but in that case, it should make that accessible using ARIA.
2. If the page already makes the mouse over accessible in some other way (e.g. uses ARIA to expose the tool tip or expose the fact that it has a menu), you will end up with redundant verbosity. So, we're punishing properly accessible websites for the sake of inaccessible ones.
3. There's no reliable way to detect a mouse over. Even where the onMouseOver attribute is used, it might be placed on an ancestor element rather than directly on the elements it handles. That would cause incorrect or spurious reporting.
4. We already have enough complaints about NVDA saying "clickable" too much and the reasons are more or less the same as the above.
5. Browsers don't expose this information via accessibility APIs. That means we'd have to use the DOM directly, which we don't use at all for any browser except IE. So, we'd need to work with all browser vendors before this could reasonably be implemented.

**Comment by jcsteh** [comment]
If we can solve issue 5, I'd be happy to have this as an option disabled by default.

**Comment by bhavyashah** [comment]
I suppose one could file a bug with Firefox and Chrome and send feedback to MS Edge, and if and when all three browsers implement the necessities, we could reconsider this ticket. Could you please do that? fisher729 
Also, due to the already mentioned reasons, I take it that this cannot currently be worked on in any way with pertinence to Project Webfix, right? @jcsteh @feerrenrut

**Comment by feerrenrut** [comment]
While a sighted user often has no visual indication that some element has a mouse over action, I'd argue that its easier (and more natural) for a sighted user to explore the webpage with the mouse, making the mouse over actions more easily discoverable. @bhavyashah correct, this will not be added to the webfix project. To get a better understanding of the priority of this issue, it would be useful if the frequency this affects users could be explored, and the impact this has in each circumstance.

**Comment by bhavyashah** [comment]
CC @Qchristensen (our best bet on all-things-mousey, as in visual or sight-oriented or heavily involving the mouse :P)

**Comment by Qchristensen** [comment]
It's a tricky one.  You might have a page which has onMouseOver events for all its links, but all they do is visually highlight the link in some way.  But you might also have a site which displays a submenu as an onMouseOver event.  If it has implemented it such that you can get to the submenu with the keyboard anyway, then you still probably don't need to know about the onMouseOver event.  There's also the point about how we get the information (5 in Jamie's comment above) which would probably need to be resolved before implementing this in any case.

**Comment by feerrenrut** [comment]
As mentioned already, at best this issue is blocked by the need to be able to get the information. However before we can apply the blocked label and a priority we should elaborate on the value this would provide. I don't think we have a solid understanding of how frequently a user misses useful information due to being unaware of the mouse over action, and what the consequence of this is.

**Comment by bhavyashah** [comment]
@jcsteh @feerrenrut Has there been any progress that you may be aware of on Point 5 of https://github.com/nvaccess/nvda/issues/6048#issuecomment-224441667, i.e. work from the end of browsers to expose the requisite information?

**Comment by jcsteh** [comment]
No progress that I'm aware of, certainly none in Firefox. And IMO, it doesn't make sense to even have this discussion until we assess the use cases/impact/value/risks; see https://github.com/nvaccess/nvda/issues/6048#issuecomment-319262996.

**Comment by Adriani90** [comment]
Actually nowadays, if a website is developed in an accessible way, mouse over elements are reported as submenu so the user of NVDA knows that he or she can expand the elements in focus mode.
If someone can provide a test webpage where we can see the negative impact of not having this, this feature, we could assess use cases much easier.

@fisher729, @Qchristensen coul you please upload a test wepage where mouse over elements are not read as submenu by NVDA and where the user does not know that there is a mouse over property on it? Thanks!

**Comment by amirsol81** [comment]
@Adriani90 Honestly I haven't seen something other than the following page from FreedomScientific which actively promotes MouseOver support for JAWS, but others might have more examples of that ilk: https://support.freedomscientific.com/Training/Surfs-Up/MouseOver.htm

**Comment by fisher729** [comment]
Hi.

They're becoming less and less frequent now, especially with the
advent of  touch devices.

**Comment by bhavyashah** [comment]
The last three comments strongly imply that (a) mouseover elements are being used less frequently nowadays, and (b) mouseover elements are reported as menus so there is some information already spoken. Combining this with https://github.com/nvaccess/nvda/issues/6048#issuecomment-224441667 and other subsequent comments mentioning the technical difficulties of fixing this, it seems like this ticket ought be given a low priority at best.

**Comment by XLTechie** [comment]
I still run into these on some corporate and government web systems on a pretty 
regular basis. When I do, they are usually in arrangements that make it 
difficult to reach by other means.

For example in one government site (it's a regional government, so very hard to 
sue for accessibility), they have a web system where you have to fill out forms, 
that frequently times out. Bad design all around.
When it does, you have to log back in, and choose your partially completed form 
by first pressing a clickable in a table.
Then, you are supposed to move to the other side of the table, and hover the 
mouse over the form number, at which point links to resume, delete, or submit 
appear.

Browse mode in NVDA gives no indication when mouse routing has caused those 
links to appear, and given the trigger needed first, it is often quite difficult 
to reach those links with a combination of mouse emulation and object nav, 
especially when you don't know if they have appeared or not.

In Jaws, this "just works", as I recently learned when helping another user fill 
one of these out. In NVDA though, it's a major pain, especially if you don't do 
it every day and don't know the tricks.

All that said, given the difficulties with implementation and user experience, I 
don't know if it's a good idea to re-open the issue.

I just wanted to provide a modern user story.

**Comment by Adriani90** [comment]
Thanks for this user story. Given the fact that it seems to work in Jaws and probably not all countries follow international web standards, I think it is reason enough to reopen.

**Comment by Adriani90** [comment]
Regarding tooltips, there is related issues #10320 and #3313.
